### PR TITLE
Bug 1903290: kubelet: refactor KUBELET_LOG_LEVEL into KubeletConfig field

### PR DESCRIPTION
**Pull request by rphillips**
**- What I did**
Takes Ben's [suggestion](https://github.com/openshift/machine-config-operator/pull/2261#discussion_r532807948) and refactors the Kubelet log level into a systemd environment file.

Obsoletes: https://github.com/openshift/machine-config-operator/pull/2261

This also changes the default of the log level to 2, since stale logs are printed into the kubelet log at level 3: [BZ1903290](https://bugzilla.redhat.com/show_bug.cgi?id=1903290).

Allows the logLevel to be configured on the KubeletConfig CRD:

```
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  name: set-kubelet-loglevel
spec:
  logLevel: 2
  machineConfigPoolSelector:
    matchLabels:
      pools.operator.machineconfiguration.openshift.io/worker: ""
```

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: templates/common/_base/files/kubelet-log-level.yaml
Comment (darkmuggle) [inline templates/common/_base/files/kubelet-log-level.yaml:6]:
```suggestion
  inline: |
      [Unit]
      Environment="KUBELET_LOG_LEVEL=3"
```

Missing indentation. Otherwise, this look much cleaner :)

**Review by darkmuggle — COMMENTED** [review]
Other than indentation, this looks good.

File: templates/common/_base/files/kubelet-log-level.yaml
Comment (rphillips) [reply templates/common/_base/files/kubelet-log-level.yaml]: thanks. fixed :)

**Comment by kikisdeliveryservice** [comment]
vpc exceeded error, reported in 4-dev

**Comment by darkmuggle** [comment]
/approve 
But I'll leave the LGTM to a member of the MCO team.

**Comment by kikisdeliveryservice** [comment]
aws/gcp are having issues across the board today would like to see passing runs before approving

File: templates/common/_base/files/kubelet-log-level.yaml
Comment (kikisdeliveryservice) [inline templates/common/_base/files/kubelet-log-level.yaml:6]: i thought level 2 was not sufficient for debugging?

Comment (rphillips) [reply templates/common/_base/files/kubelet-log-level.yaml:6]: Problem is that level 3 is logging dead pod logs into the kubelet log. Instead of chasing all level 3 logs... I'm proposing just setting it to level 2. This new approach with a kubelog.conf would allow us to manage the kubelet log level within the kubeletconfig perhaps.

[Log Line](https://github.com/openshift/origin/blob/release-4.4/vendor/k8s.io/kubernetes/pkg/kubelet/status/status_manager.go#L434)

Comment (rphillips) [reply templates/common/_base/files/kubelet-log-level.yaml:6]: I'm open to changing the log line to be 4, but it seems like whack-a-mole.

Comment (kikisdeliveryservice) [reply templates/common/_base/files/kubelet-log-level.yaml:6]: If there's consensus that level work works between the experts (ie you and @sjenning) I'm ok with it, just want to make sure this isn't going to hurt us later.

@sjenning wdyt?

**Comment by kikisdeliveryservice** [comment]
uh since ci is still underwater could you update: https://github.com/openshift/machine-config-operator/blob/master/docs/KubeletConfigDesign.md

**Comment by kikisdeliveryservice** [comment]
unit and verify failures don't look to be flakes..

**Comment by rphillips** [comment]
Updated to fix the unit tests and allow support for only setting the logLevel

**Comment by rphillips** [comment]
Vpc limit

/retest

**Comment by kikisdeliveryservice** [comment]
@rphillips brought this up again in 4-dev no aws resolution yet :frowning_face:

**Comment by kikisdeliveryservice** [comment]
It's getting late but I'll test this on a cluster tomorrow.

File: templates/common/_base/files/kubelet-log-level.yaml
Comment (sjenning) [reply templates/common/_base/files/kubelet-log-level.yaml:6]: @kikisdeliveryservice I'm on board with this

**Review by kikisdeliveryservice — APPROVED** [review]
Tested & seems to work ok - made kubelet config applied, then changed kubelet logging seemed to change appropriately.

**Comment by sjenning** [comment]
still hitting VPC limits issues :disappointed:

File: pkg/controller/kubelet-config/helpers.go
Comment (sjenning) [inline pkg/controller/kubelet-config/helpers.go:29]: This is creating some weird whitespace issues
```
sh-4.4# cat 20-logging.conf 

[Service]
Environment="KUBELET_LOG_LEVEL=3"
	sh-4.4#
```

Comment (sjenning) [reply pkg/controller/kubelet-config/helpers.go:29]: actually, this is more than a nit because if I go from having no `KubeletConfig` to having one that specifies log-level 2 (the default), I get an unexpected rollout because the default `20-logging.conf` doesn't match the rendered one.

Comment (kikisdeliveryservice) [reply pkg/controller/kubelet-config/helpers.go:29]: Given that this is a be probably should just fix it now instead of creating another bz to fix later.

**Comment by kikisdeliveryservice** [comment]
Since Seth found an issue, I think we should fix it first then merge

/hold

File: pkg/controller/kubelet-config/helpers.go
Comment (rphillips) [reply pkg/controller/kubelet-config/helpers.go]: fixed!

**Comment by rphillips** [comment]
/hold cancel

fixed the issue.

**Comment by sjenning** [comment]
Trying it out now. Will lgtm when verified.

**Pull request merged**
